services/concept_insights/ci.js

```diff
@@ -0,0 +1,422 @@
+            corpus_name = this.corpus.cname;
+          } else {
+            console.log(this.corpus);
+            corpus_name = this.corpus.publiccorpus;
+            account_id = 'public';
+          }        
+          var document_name = config.docname;
+
+          node.status({fill:"blue",shape:"ring",text:"requesting..."});
+
+          //Check for document or corpus mode
+          if (config.mode === 'document') {
+
+            if (!document_name) {
+              node.status({});
+              var message = 'No document name specified';
+              return node.error(message, msg);
+            }
+
+            var params = {
+              id: '/corpora/'+account_id+'/'+corpus_name+'/documents/'+document_name
+            }
+
+            switch (config.docapicall) {
+              case 'annotations':
+                params.id = params.id+'/annotations';
+                concept_insights.corpora.getDocumentAnnotations(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relatedconcepts':
+                concept_insights.corpora.getRelatedConcepts(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relationscores':
+              params.concepts = msg.concepts;
+                concept_insights.corpora.getRelationScores(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+            }
+          } else {
+            //Corpus mode
+            var params = {
+              corpus: '/corpora/'+account_id+'/'+corpus_name
+            }
+            switch (config.corpusapicall) {
+              case 'statistics':
+                concept_insights.corpora.getCorpusStats(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relatedconcepts':
+                concept_insights.corpora.getRelatedConcepts(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relationscores':
+                params.concepts = msg.concepts;
+                concept_insights.corpora.getRelationScores(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'conceptualsearch':
+                if (msg.concepts) {
+                  params.ids = msg.concepts;
+                  concept_insights.corpora.getRelatedDocuments(params, function(err,res) {
+                    handleWatsonCallback(node,msg,err,res);
+                  });
+                } else {
+                  node.status({});
+                  var message = 'No concepts exist on the msg object';
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 69.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2388498338/issues/source?bid=3225115&fileBranchId=3228814#l131)